pkg/machines/components/storagePools/storageVolumeDelete.jsx

```diff
@@ -0,0 +1,127 @@
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showModal: false,
+        };
+
+        this.close = this.close.bind(this);
+        this.open = this.open.bind(this);
+        this.storageVolumeListDelete = this.storageVolumeListDelete.bind(this);
+    }
+
+    close() {
+        this.setState({ showModal: false });
+    }
+
+    open() {
+        this.setState({ showModal: true });
+    }
+
+    storageVolumeListDelete({force = false}) {
+        const { volumes, usedBy, storagePool, resetSelection } = this.props;
+        const anyVolumeUsed = volumes.some(volume => usedBy[volume].length == 0);
+
+        if (anyVolumeUsed || force) {
+            cockpit.all(volumes.map(volume =>
+                deleteStorageVolume({
+                    poolName: storagePool.name,
+                    connectionName: storagePool.connectionName,
+                    volumePath: [storagePool.path, volume].join('/'),
+                })
+            ))
+                    .fail(exc => {
+                        this.props.deleteErrorHandler(_("Storage Volumes could not be deleted"), exc.message);
+                        if (force)
+                            this.close();
+                    })
+                    .then(() => {
+                        resetSelection();
+                        if (force)
+                            this.close();
+                    });
+        } else {
+            this.open();
+        }
+    }
+
+    render() {
+        const { volumes, usedBy } = this.props;
+        const volCount = volumes.length;
+
+        if (volCount == 0)
+            return null;
+
+        return (
+            <div>
+                <Button className='storage-volumes-actions' id='storage-volumes-delete' bsStyle='danger' onClick={this.storageVolumeListDelete}>
+                    {`${_("Delete")} ${volCount} ${volCount > 1 ? 'volumes' : 'volume'}`}
+                </Button>
+                { this.state.showModal && <Modal show onHide={this.close} id='storage-volumes-delete-modal'>
+                    <Modal.Header>
+                        <Modal.CloseButton onClick={this.close} />
+                        <Modal.Title> {'Delete Storage Volumes'} </Modal.Title>
+                    </Modal.Header>
+                    <Modal.Body>
+                        <div>
+                            {_("The following Storage Volumes are currently in use.")}
+                            <ul>
+                                { volumes.map(volume => <li key={volume}> {volume} {_("is used by")} {usedBy[volume].join(', ')} </li>) }
+                            </ul>
+                        </div>
+                    </Modal.Body>
+                    <Modal.Footer>
+                        <span className='idle-message'>
+                            <i className='pficon pficon-warning-triangle-o' />
+                            <span>{_("Deleting volumes used by running VMs might destroy them")}</span>
```

Comment:
I am ok with the option to not allow deleting volumes that are used by VMs, running or not.
But the warning isn't false, `will destroy them` is not always true. For example for disk on local filesystem, deleting the disk file while the VM is using it, will not result in destroying the VM. The VM still has reference to file inode, and until it will shutdown it will be using the deleted disk without problem.

Reply:
But, I think it makes sense to forbid deletion of used disks in cockpit in order to make the UI simpler if you agree as well.